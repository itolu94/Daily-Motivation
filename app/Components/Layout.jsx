import React, {Component} from 'react';
import Header from './Content/Header.jsx';
import Registration from './Content/Pages/Registration.jsx';
import DeleteAccount from './Content/Pages/DeleteAccount.jsx';
import Contact from './Content/Pages/Contact.jsx';
import Footer from './Content/Footer.jsx';

export default class Layout extends Component {
    constructor(){
        super();
        this.state={
            page: ''
        };
        this.changePage =  this.changePage.bind(this);
    }

    changePage(e){
        this.setState({page: e.target.name});
    }
    renderPage(){
        switch(this.state.page){
            case "registration":
                return <Registration/>
            case "delete-account":
                return <DeleteAccount/>
            case "contact":
                return <Contact/>
            default:
                return <Registration/>
        }
    }
    render(){
        return(
            <div id='content' >
                <Header changePage={this.changePage} />
                {this.renderPage()}
                <Footer/>
            </div>
        )
    }
}