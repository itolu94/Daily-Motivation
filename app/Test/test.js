import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import { shallow } from 'enzyme'
import Layout from '../Components/Layout.jsx';
import renderer from 'react-test-renderer';


test("Layout component", () => {
   const component = shallow(<Layout/>)
   expect(component).toMatchSnapshot();
});