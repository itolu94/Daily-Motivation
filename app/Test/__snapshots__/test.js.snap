// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component 1`] = `
<div
  id="content"
>
  <Header
    changePage={[Function]}
  />
  <Registration />
  <Footer />
</div>
`;

exports[`Layout should display registration component 1`] = `
<div
  id="content"
>
  <nav
    id="header"
  >
    <div
      className="nav-wrapper"
    >
      <a
        className="brand-logo center"
        href="#"
      >
        Daily-Motivation
      </a>
      <ul
        className="left hide-on-med-and-down"
        id="nav-mobile"
      >
        <li>
          <a
            href="#"
            name="registration"
            onClick={[Function]}
          >
            Registration
          </a>
        </li>
        <li>
          <a
            href="#"
            name="delete-account"
            onClick={[Function]}
          >
            Delete Account
          </a>
        </li>
        <li>
          <a
            href="#"
            name="contact"
            onClick={[Function]}
          >
            Contact
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container"
    id="registration-div"
  >
    <div
      className="row pages-div"
    >
      <div
        className="col m8 offset-m2 center-align"
      >
        <h3
          id="registration-header"
        >
           Sign up to receive free motivational quotes to start your day
        </h3>
        <p>
          
        </p>
        <form
          onSubmit={[Function]}
        >
          <input
            name="userName"
            onChange={[Function]}
            placeholder="First Name"
            required={true}
            type="text"
            value=""
          />
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            required={true}
            type="text"
            value=""
          />
          <input
            name="phoneNumber"
            onChange={[Function]}
            placeholder="Phone Number"
            required={true}
            type="text"
            value=""
          />
          <input
            className=""
            type="submit"
          />
        </form>
      </div>
    </div>
  </div>
  <div
    id="footer"
  >
    Â© 2018 Tolu Idowu
  </div>
</div>
`;
